### PR TITLE
misra shadow: resolve 10.1 rule violations

### PR DESCRIPTION
**Pull request by nateglims**
*Description of changes:*
* Add unsigned annotation to flag literals
* Change variable used for shifting to unsigned
* Explicitly cast enum value used in arithmetic expressions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/aws/shadow/src/aws_iot_shadow_api.c [hunk 1]
```diff
@@ -349,7 +349,7 @@ static AwsIotShadowError_t _setCallbackCommon( IotMqttConnection_t mqttConnectio
     }
 
     /* Check parameters. */
-    status = _validateThingNameFlags( ( _shadowOperationType_t ) ( type + SHADOW_OPERATION_COUNT ),
+    status = _validateThingNameFlags( ( _shadowOperationType_t ) ( ( int32_t ) type + SHADOW_OPERATION_COUNT ),
```
Comment (cobusve) [inline libraries/aws/shadow/src/aws_iot_shadow_api.c]: Is your intent to typecast "type" and then let the Integer promotion take care of the rest, or is your intent to cast the result of the opertation? I think brackets make the intent clearer in these cases.

**Comment by cobusve** [comment]
Also you will need to refresh this branch is out of date now

File: libraries/aws/shadow/src/aws_iot_shadow_api.c (hunk 1 above)
Comment (nateglims) [reply libraries/aws/shadow/src/aws_iot_shadow_api.c]: The specific violation is that we are doing arithmetic with an enum (`type`).

Comment (cobusve) [reply libraries/aws/shadow/src/aws_iot_shadow_api.c]: Then I expect (( int32_t) type ) + SHADOW_OPERATION_COUNT

**Pull request merged**